### PR TITLE
Use concurrently and wait-on for fast incremental builds

### PR DESCRIPTION
**Pull request by Sacro**

### CODE REVIEW

**Comment by Sacro** [comment]
Maybe we could consider removing `ts-node` as it's only used in `start` now.

File: nodemon.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
-  "watch": ["src"],
-  "ext": "ts",
-  "ignore": ["src/**/*.spec.ts"],
-  "exec": "ts-node -r tsconfig-paths/register src/main.ts"
+  "watch": ["dist"],
+  "ext": "js",
+  "exec": "node -r tsconfig-paths/register -r ts-node/register dist/main.js"
```
Comment (kamilmysliwiec) [inline nodemon.json]:
```
 -r tsconfig-paths/register -r ts-node/register 
```
Why are these required?

Comment (Sacro) [reply nodemon.json]: > ```
>  -r tsconfig-paths/register -r ts-node/register 
> ```
> Why are these required?

`tsconfig-paths` is used when people use `paths` in their `tsconfig.json` to map things in.

`ts-node/register` can be dropped

Comment (kamilmysliwiec) [reply nodemon.json]: > tsconfig-paths is used when people use paths in their tsconfig.json to map things in.

Are you sure that it will still work after compilation? `tsconfig-paths` is used with combination with the `ts-node`

**Comment by kamilmysliwiec** [comment]
> Maybe we could consider removing ts-node as it's only used in start now.

We shouldn't because it's being used by `start`.

**Comment by Sacro** [comment]
> > Maybe we could consider removing ts-node as it's only used in start now.
> 
> We shouldn't because it's being used by `start`.

Right, but we could move that off `ts-node` surely?

**Comment by kamilmysliwiec** [comment]
If you run `npm run start:dev` in a row (2 times), it won't start your app. Since TS caches the build info in a dedicated file now, once you try to rerun `start:dev` (without changing a line of code), it won't trigger the build = `wait-on` will hang out the process.

**Comment by Sacro** [comment]
> If you run `npm run start:dev` in a row (2 times), it won't start your app. Since TS caches the build info in a dedicated file now, once you try to rerun `start:dev` (without changing a line of code), it won't trigger the build = `wait-on` will hang out the process.

It's not outputting to "dist" when you delete it, so it makes sense to not delete it and keep the `.tsbuildinfo` file in its default location.

**Comment by kamilmysliwiec** [comment]
LGTM
